force-app/main/default/classes/RollupRecord.cls

```diff
@@ -124,25 +167,253 @@ public with sharing class RollupRecord {
         return isModified;
     }
 
-    public SObject getSObject() {
-        SObject sObj = recordId.getSObjectType().newSObject(recordId);
+    @TestVisible
+    private void incrementProcessed() {
+        totalProcessed++;
+    }
 
+    @TestVisible
+    private void processPresent(Date presentDate) {
+        incrementPresent();
+
+        if (lastPresentDate == null || presentDate > lastPresentDate) {
+            lastPresentDate = presentDate;
+            processConsecutiveAbsences();
+        }
+
+        consecutiveCount = consecutiveAbsenceDates.size();
+    }
+
+    @TestVisible
+    private void processAbsent(Date absentDate) {
+        incrementAbsent();
+
+        if (lastPresentDate == null || lastPresentDate < absentDate) {
+            consecutiveAbsenceDates.add(absentDate);
+            consecutiveCount = consecutiveAbsenceDates.size();
+        }
+    }
+
+    @TestVisible
+    private void processLastServiceDate(Date deliveryDate) {
+        if (lastServiceDateField == null) {
+            return;
+        }
+
+        lastServiceDate = lastServiceDate == null ||
+            lastServiceDate < deliveryDate
+            ? deliveryDate
+            : lastServiceDate;
+    }
+
+    private void processConsecutiveAbsences() {
```

Comment:
Do we need this method? Can we not add the early return to updateConsecutiveAbsenceDates() and use that method instead or did I miss something

Reply:
Great catch, must have been sleeping at the wheel lol.